pallets/thea/src/tests.rs

```diff
@@ -0,0 +1,306 @@
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Tests for pallet-ocex.
+
+use crate::{mock::*, *};
+use bls_primitives::Pair;
+use frame_support::{assert_err, assert_ok};
+use sp_core::Pair as CorePair;
+use sp_runtime::DispatchError::BadOrigin;
+
+const WELL_KNOWN: &str = "bottom drive obey lake curtain smoke basket hold race lonely fit walk";
+
+const PAYLOAD: [u8; 10_485_760] = [u8::MAX; 10_485_760];
+
+fn any_id() -> <Test as Config>::TheaId {
+	<Test as Config>::TheaId::decode(&mut [1u8; 96].as_ref()).unwrap()
+}
+
+fn any_signature() -> <Test as Config>::Signature {
+	<Test as Config>::Signature::decode(&mut [1u8; 48].as_ref()).unwrap()
+}
+
+fn set_200_validators(network: u8) -> [Pair; 200] {
+	let mut validators = Vec::with_capacity(200);
+	for i in 0..200 {
+		validators
+			.push(Pair::generate_with_phrase(Some(format!("{}//{}", WELL_KNOWN, i).as_str())).0);
+	}
+	let mut bv: BoundedVec<<Test as Config>::TheaId, <Test as Config>::MaxAuthorities> =
+		BoundedVec::with_max_capacity();
+	validators
+		.clone()
+		.into_iter()
+		.for_each(|v| bv.try_push(v.public().into()).unwrap());
+	<Authorities<Test>>::insert(network, 0, bv);
+	validators
+		.try_into()
+		.unwrap_or_else(|_| panic!("Could not convert validators to array"))
+}
+
+fn message_for_nonce(nonce: u64) -> Message {
+	Message {
+		block_no: u64::MAX,
+		nonce,
+		data: [255u8; 576].into(), //10 MB
+		network: 0u8,
+		is_key_change: false,
+		validator_set_id: 0,
+		validator_set_len: 200,
+	}
+}
+
+#[test]
+fn test_update_network_pref_bad_origin() {
+	new_test_ext().execute_with(|| {
+		assert_err!(
+			Thea::update_network_pref(RuntimeOrigin::root(), any_id(), 0, any_signature()),
+			BadOrigin
+		);
+	})
+}
+
+#[test]
+fn test_update_network_pref_success() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(Thea::update_network_pref(RuntimeOrigin::none(), any_id(), 0, any_signature()));
+	})
+}
+
+// following test does:
+// 1. creates and inserts 200 validators as authorities for network 0
+// 2. creates 200 messages signed by each of 200 validators in turn
+// 3. submits them sequentially
+// 4. validates runtime accepts it successfuly
+#[test]
+fn test_lots_of_incoming_messages_with_200_validators_ok() {
+	new_test_ext().execute_with(|| {
+		// 200 validators
+		let validators = set_200_validators(0);
```

Comment:
I'd suggest to add assertion here to be sure we didn't receive regression on change `set_200_validators` function since below assertions happening only in loop and could be missed.
```suggestion
		let validators = set_200_validators(0);
		assert_eq!(validators.len(), 200);
```